File: frontend/src/types/scheduler.ts
export interface WebhookConfig {
  id: string
  name: string
  url: string
  enabled: boolean
  platform: string
  secret: string
  skipSSL: boolean
}

export interface SchedulerTask {
  id: string
  name: string
  enabled: boolean
  cronExpr: string
  webhooks: WebhookConfig[]
  projectId: number
  productId: number
  projectName: string
  productName: string
  statusFilter: string
  reportType: string
  keyword: string
  externalInfo: string
  lastRunAt: string | null
  lastRunStatus: string
  createdAt: string
  updatedAt: string
}

export interface WebhookResult {
  webhookId: string
  webhookName: string
  webhookUrl: string
  success: boolean
  statusCode: number
  error: string
}

export interface TaskExecutionLog {
  id: string
  taskId: string
  taskName: string
  startedAt: string
  finishedAt: string | null
  status: 'running' | 'success' | 'partial' | 'failed'
  bugTotal: number
  highSeverity: number
  assigneeCount: number
  webhookResults: WebhookResult[]
  error: string
}

export interface AssigneeBugStats {
  assignee: string
  account: string
  total: number
  highSeverity: number
  fatal: number
  serious: number
  moderate: number
  minor: number
}

export interface BugReport {
  title: string
  timestamp: string
  projectName: string
  total: number
  highSeverity: number
  statusBreakdown: Record<string, number>
  details: AssigneeBugStats[]
  message: string
}

export const CRON_PRESETS = [
  { label: '每天 9:00', expr: '0 9 * * *' },
  { label: '工作日 9:00', expr: '0 9 * * 1-5' },
  { label: '每周一 9:00', expr: '0 9 * * 1' },
  { label: '每8小时', expr: '0 */8 * * *' },
  { label: '每30分钟', expr: '*/30 * * * *' },
] as const

export const STATUS_OPTIONS = [
  { label: '活跃 (active)', value: 'active' },
  { label: '已解决 (resolved)', value: 'resolved' },
  { label: '已关闭 (closed)', value: 'closed' },
  { label: '全部', value: 'all' },
] as const

export const REPORT_TYPE_OPTIONS = [
  { label: 'Bug 分布报告', value: 'bug', icon: '🐛', color: '#EF4444' },
  { label: '需求进度播报', value: 'requirement', icon: '📋', color: '#4F6BF6' },
  { label: '任务进度播报', value: 'task', icon: '✅', color: '#22C55E' },
] as const

export interface AssigneeStoryStats {
  assignee: string
  account: string
  total: number
  active: number
  changed: number
  closed: number
  resolved: number
  accepted: number
}

export interface RequirementReport {
  title: string
  timestamp: string
  projectName: string
  productName: string
  total: number
  statusBreakdown: Record<string, number>
  details: AssigneeStoryStats[]
  message: string
}

export interface TaskProgressStats {
  assignee: string
  account: string
  total: number
  wait: number
  doing: number
  done: number
  paused: number
  cancelled: number
  estimate: number
  consumed: number
  progress: number
}

export interface TaskProgressReport {
  title: string
  timestamp: string
  projectName: string
  productName: string
  total: number
  statusBreakdown: Record<string, number>
  totalEstimate: number
  totalConsumed: number
  overallProgress: number
  details: TaskProgressStats[]
  message: string
}

File: backend/core/models/scheduler.go
package models

import "time"

type WebhookConfig struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	URL      string `json:"url"`
	Enabled  bool   `json:"enabled"`
	Platform string `json:"platform"` // generic | lanxin
	Secret   string `json:"secret"`   // 蓝信加签密钥
	SkipSSL  bool   `json:"skipSSL"`  // 跳过SSL证书验证
}

type SchedulerTask struct {
	ID            string          `json:"id"`
	Name          string          `json:"name"`
	Enabled       bool            `json:"enabled"`
	CronExpr      string          `json:"cronExpr"`
	Webhooks      []WebhookConfig `json:"webhooks"`
	ProjectID     int             `json:"projectId"`
	ProductID     int             `json:"productId"`
	ProjectName   string          `json:"projectName"`
	ProductName   string          `json:"productName"`
	StatusFilter  string          `json:"statusFilter"`
	ReportType    string          `json:"reportType"` // bug | requirement | task | bug-aging
	AgingDays     int             `json:"agingDays"`  // bug-aging 超时天数阈值，默认 7
	Keyword       string          `json:"keyword"`
	ExternalInfo  string          `json:"externalInfo"`
	LastRunAt     *time.Time      `json:"lastRunAt"`
	LastRunStatus string          `json:"lastRunStatus"`
	CreatedAt     time.Time       `json:"createdAt"`
	UpdatedAt     time.Time       `json:"updatedAt"`
}

type TaskExecutionLog struct {
	ID             string           `json:"id"`
	TaskID         string           `json:"taskId"`
	TaskName       string           `json:"taskName"`
	StartedAt      time.Time        `json:"startedAt"`
	FinishedAt     *time.Time       `json:"finishedAt"`
	Status         string           `json:"status"`
	BugTotal       int              `json:"bugTotal"`
	HighSeverity   int              `json:"highSeverity"`
	AssigneeCount  int              `json:"assigneeCount"`
	WebhookResults []WebhookResult  `json:"webhookResults"`
	Error          string           `json:"error,omitempty"`
}

type WebhookResult struct {
	WebhookID   string `json:"webhookId"`
	WebhookName string `json:"webhookName"`
	WebhookURL  string `json:"webhookUrl"`
	Success     bool   `json:"success"`
	StatusCode  int    `json:"statusCode,omitempty"`
	Error       string `json:"error,omitempty"`
}

type AssigneeBugStats struct {
	Assignee     string `json:"assignee"`
	Account      string `json:"account"`
	Total        int    `json:"total"`
	HighSeverity int    `json:"highSeverity"`
	Fatal        int    `json:"fatal"`
	Serious      int    `json:"serious"`
	Moderate     int    `json:"moderate"`
	Minor        int    `json:"minor"`
}

type BugReport struct {
	Title           string                    `json:"title"`
	Timestamp       string                    `json:"timestamp"`
	ProjectName     string                    `json:"projectName"`
	Total           int                       `json:"total"`
	HighSeverity    int                       `json:"highSeverity"`
	StatusBreakdown map[string]int            `json:"statusBreakdown"`
	Details         []AssigneeBugStats        `json:"details"`
	Message         string                    `json:"message"`
}

type CronDB struct {
	Tasks []SchedulerTask    `json:"tasks"`
	Logs  []TaskExecutionLog `json:"logs"`
}

type AssigneeStoryStats struct {
	Assignee   string `json:"assignee"`
	Account    string `json:"account"`
	Total      int    `json:"total"`
	Active     int    `json:"active"`
	Changed    int    `json:"changed"`
	Closed     int    `json:"closed"`
	Resolved   int    `json:"resolved"`
	Accepted   int    `json:"accepted"`
}

type RequirementReport struct {
	Title           string                 `json:"title"`
	Timestamp       string                 `json:"timestamp"`
	ProjectName     string                 `json:"projectName"`
	ProductName     string                 `json:"productName"`
	Total           int                    `json:"total"`
	StatusBreakdown map[string]int         `json:"statusBreakdown"`
	Details         []AssigneeStoryStats   `json:"details"`
	Message         string                 `json:"message"`
}

type TaskProgressStats struct {
	Assignee   string  `json:"assignee"`
	Account    string  `json:"account"`
	Total      int     `json:"total"`
	Wait       int     `json:"wait"`
	Doing      int     `json:"doing"`
	Done       int     `json:"done"`
	Paused     int     `json:"paused"`
	Cancelled  int     `json:"cancelled"`
	Estimate   float64 `json:"estimate"`
	Consumed   float64 `json:"consumed"`
	Progress   float64 `json:"progress"`
}

type TaskProgressReport struct {
	Title           string              `json:"title"`
	Timestamp       string              `json:"timestamp"`
	ProjectName     string              `json:"projectName"`
	ProductName     string              `json:"productName"`
	Total           int                 `json:"total"`
	StatusBreakdown map[string]int      `json:"statusBreakdown"`
	TotalEstimate   float64             `json:"totalEstimate"`
	TotalConsumed   float64             `json:"totalConsumed"`
	OverallProgress float64             `json:"overallProgress"`
	Details         []TaskProgressStats `json:"details"`
	Message         string              `json:"message"`
}

type BugAgingItem struct {
	ID         int    `json:"id"`
	Title      string `json:"title"`
	Severity   string `json:"severity"`
	OpenedDate string `json:"openedDate"`
	DaysOpen   int    `json:"daysOpen"`
}

type AssigneeBugAgingStats struct {
	Assignee string           `json:"assignee"`
	Account  string           `json:"account"`
	Total    int              `json:"total"`
	Bugs     []BugAgingItem   `json:"bugs"`
}

type BugAgingReport struct {
	Title       string                  `json:"title"`
	Timestamp   string                  `json:"timestamp"`
	ProjectName string                  `json:"projectName"`
	Total       int                     `json:"total"`
	AgingDays   int                     `json:"agingDays"`
	Details     []AssigneeBugAgingStats `json:"details"`
	Message     string                  `json:"message"`
}
